bigtable/integration_test.go

```diff
@@ -2263,8 +2263,12 @@ func TestIntegration_InstanceAdminClient_AppProfile(t *testing.T) {
 	}
 
 	// App Profile list should contain default, app_profile1
-	if got, want := len(profiles), 2; got != want {
-		t.Fatalf("Initial app profile list len: %d, want: %d", got, want)
+	wantProfiles := map[string]struct{}{"default": {}, profileID: {}}
+	for _, profile := range profiles {
```

Comment:
This took me a moment to figure out what's going on (without reading the file). The more verbose version might be:

``` golang
equalProfiles := func(a, b map[string]struct{}) {
  if len(a) != len(b) {
    return false
  }
  for k, _ := range a {
    if _, ok := b[k]; !ok {
      return false
    }
  }
  return true
}

wantProfiles := map[string]struct{}{"default": {}, profileID: {}}
if !equalProfiles(wantProfiles, profiles) {
  t.Fatalf("...")
}
```

But if `equalProfiles` weren't useful anywhere else, this more concise version is probably better.

Reply:
So this is the interesting bit, it isn't checking for equality. 

The test creates profiles ('a', 'b')
The live service has existing profile ('c')

we need to verify all profiles we created exist, but we don't care about 'c'.

That said, I can certainly
1) add a comment explaining this
2) extract the slightly odd code to a func as you did with a name like "ensureProfiles" or something of the sort.

Reply:
Extracting the check into a function seems useful and a comment, if only for the tired, future reader. :-)

Reply:
I restructured this. I am not sure that brought much clarity, but I could be wrong. I did add a comment explaining what I am checking here though. lmk what you think.

Reply:
I'm a sucker for intention revealing code and adding the function makes a big difference for me. ^^;